telemetry/telemetry.go

```diff
@@ -0,0 +1,116 @@
+package telemetry
+
+/*
+  methods based on Consul telemetry:
+    https://github.com/hashicorp/consul/blob/main/lib/telemetry.go
+*/
+
+import (
+	"context"
+	"log"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/armon/go-metrics"
+	"github.com/hashicorp/go-multierror"
+
+	"github.com/hashicorp/consul-template/config"
+)
+
+// MetricsHandler provides an http.Handler for displaying metrics.
+type MetricsHandler interface {
+	DisplayMetrics(resp http.ResponseWriter, req *http.Request) (interface{}, error)
+	Stream(ctx context.Context, encoder metrics.Encoder)
+}
+
+type Telemetry struct {
+	Handler  MetricsHandler
+	mu       sync.Mutex
+	cancelFn context.CancelFunc
+}
+
+func (tel *Telemetry) Stop() {
+	tel.mu.Lock()
+	defer tel.mu.Unlock()
+
+	if tel.cancelFn != nil {
+		tel.cancelFn()
+	}
+}
+
+func ConfigureSinks(cfg *config.TelemetryConfig, memSink metrics.MetricSink) (metrics.FanoutSink, error) {
+	metricsConf := metrics.DefaultConfig(cfg.MetricsPrefix)
+	metricsConf.EnableHostname = !cfg.DisableHostname
+	metricsConf.FilterDefault = cfg.FilterDefault
+	metricsConf.AllowedPrefixes = cfg.AllowedPrefixes
+	metricsConf.BlockedPrefixes = cfg.BlockedPrefixes
+
+	var sinks metrics.FanoutSink
+	var errors error
```

Comment:
i suggest using a multierror here so that at the end of the function we don't have to check for errors length and we can just do `errors.ErrorOrNil()`
```suggestion
	var errors *multierror.Error
```

Reply:
I didn't know about it until recently, and it is indeed very helpful!
Code updated to include suggestion.